src/newcharacter.cpp

```diff
@@ -380,6 +381,15 @@ int player::create(character_type type, std::string tempname)
     ret_null = item("null", 0);
     weapon = ret_null;
 
+    //Learn recipes
+    for( auto &cur_recipe : recipe_dict ) {
+        if ( !cur_recipe->autolearn && g->u.has_recipe_requirements(cur_recipe) &&
+            cur_recipe->ident.find("uncraft") == std::string::npos &&
+            !(g->u.learned_recipes.find(cur_recipe->ident) != g->u.learned_recipes.end()) ) {
+
+            g->u.learn_recipe( (recipe *)cur_recipe );
```

Comment:
You can remove the `g->u.` *here* (in this function), the context is already that of `g->u`. The code above (`weapon`, `scent`, ...) accesses `this` directly, it's a bit confusing if this here does use `g->u`.